tests/cases/conformance/parser/ecmascript6/Iterators/parserForOfStatement22.ts

```diff
@@ -0,0 +1,4 @@
+// @target: esnext

```

Comment:
we should also have tests for `for (async in ...` and `for await (async of ...` as pointed out by @Zzzen. If those tests already exist, just point them out, no need to duplicate them.